contracts/pool/src/tests/liquidity.rs

```diff
@@ -1471,3 +1490,213 @@ fn provide_liquidity_and_autostake_should_fail_when_liquidity_is_not_enough_for_
         &true,
     );
 }
+
+#[test]
+fn withdraw_liquidity_with_auto_unstake() {
+    let env = Env::default();
+    env.mock_all_auths();
+    env.cost_estimate().budget().reset_unlimited();
+
+    let admin = Address::generate(&env);
+
+    let mut token1 = deploy_token_contract(&env, &admin);
+    let mut token2 = deploy_token_contract(&env, &admin);
+    let reward_token = deploy_token_contract(&env, &admin);
+
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+    }
+
+    let user = Address::generate(&env);
+    let stake_manager = Address::generate(&env);
+    let stake_owner = Address::generate(&env);
+
+    let pool = deploy_liquidity_pool_contract(
+        &env,
+        None,
+        (&token1.address, &token2.address),
+        0,
+        None,
+        None,
+        None,
+        stake_manager.clone(),
+        stake_owner,
+    );
+
+    let share_token_address = pool.query_share_token_address();
+    let token_share = token_contract::Client::new(&env, &share_token_address);
+
+    let result = pool.query_pool_info();
+    let stake = stake_contract::Client::new(&env, &result.stake_address);
+
+    stake.create_distribution_flow(&stake_manager, &reward_token.address);
+    let reward_amount = 1_000_000_000_000_000;
+    reward_token.mint(&stake_manager, &reward_amount);
+    let distribution_duration = 10_000;
+    stake.fund_distribution(
+        &stake_manager,
+        &0,
+        &distribution_duration,
+        &reward_token.address,
+        &reward_amount,
+    );
+
+    token1.mint(&user, &1_000_000_000_000_000);
+    token2.mint(&user, &1_000_000_000_000_000);
+    let stake_timestamp = env.ledger().timestamp();
+    pool.provide_liquidity(
+        &user,
+        &Some(1_000_000_000_000_000),
+        &Some(1_000_000_000_000_000),
+        &Some(1_000_000_000_000_000),
+        &Some(1_000_000_000_000_000),
+        &None,
+        &None::<u64>,
+        &true,
+    );
+
+    let initial_stake = stake.query_staked(&user);
+    assert_eq!(
+        initial_stake,
+        StakedResponse {
+            stakes: vec![
+                &env,
+                Stake {
+                    stake: 999_999_999_999_000,
+                    stake_timestamp
+                }
+            ]
+        }
+    );
+    assert_eq!(
+        initial_stake.stakes.get(0).unwrap().stake,
+        999_999_999_999_000
+    );
+    assert_eq!(
+        initial_stake.stakes.get(0).unwrap().stake_timestamp,
+        stake_timestamp
+    );
+
+    let share_amount = 500_000_000_000_000;
+    let min_a = 500_000_000_000_000;
+    let min_b = 500_000_000_000_000;
+    let auto_unstake_info = AutoUnstakeInfo {
+        stake_amount: 999_999_999_999_000,
+        stake_timestamp,
+    };
```

Comment:
Can you assign those amounts somewhere? I'm jumping up and down constantly trying to figure out how exactly did you unstake and what part of the total liquidity it should be.